pkg/arvo/lib/quiz.hoon

```diff
@@ -0,0 +1,180 @@
+|_  [eny=@uv runs=@ud]
+++  give
+  |$  [sam]
+  $-([@ud _og] sam)
+++  check
+  |*  [vax=vase give=(unit (give)) alts=(unit $-(vase (list vase)))]
+  =?  runs  =(0 runs)  100
+  =+  size=1
+  =+  sax=(slot 6 vax)
+  =+  run-i=0
+  =+  rng=~(. og eny)
+  =+  drop=0
+  |-  ^-  ?
+  ?:  =(run-i runs)
+      ~&  [success-runs+run-i drops+drop]
+      %.y
+  =/  sam=vase
+    ?~  give
+      (~(fill quiz [size rng]) sax)
+    [p=p.sax q=(u.give size rng)]
+  =+  res=(slum q.vax q.sam)
```

Comment:
I'm working with this library and am wondering if `+mink` works where `+slum` is used here. That would have the benefit of allowing the library to test gates which can crash, and ultimately allow one to run many tests and then print the failing ones along with their inputs at the end.

Reply:
I'm not a big fan of the idea of keeping going after a failure is found, partly because runtimes can often increase with size. I prefer just returning (and shrinking) as soon an error is encountered.

That being so, we can probably get away with just using `~|`

```
=+  ^=  res
~| [%quiz-crash-on-input q.sam]
(slum q.vax q.sam)
```

Reply:
actually this won't work well, since the program would still crash. virtualizing with `mink` (or `mong` in this case, which just wraps `mink`) makes sense.